File: euclid/euclid.pas
{* euclid.pas *}

program euclid;

var a,b,result:integer;


{* Greatest Common Divisor Function*}
function GCD(a, b : integer) : integer;

var t, r : integer;

begin
 if (a < b) then
    begin
      t := a; a := b; b := t
    end;

 repeat
    r := a mod b; {* modulus *}
    a := b;
    b := r
 until r = 0;

 GCD := a;
end;

{* Main Function *}
begin

writeln('Greatest Common Divisor (GCD)');
writeln('using Euclid Algorithm');
write('Enter number A:'); readln(a);
write('Enter number B:'); readln(b);

result:= GCD(a,b);

writeln(' GCD = ', result);

end.

File: euclid/euclid.f
*     euclid.f (FORTRAN 77)
*     Find greatest common divisor using the Euclidean algorithm

      PROGRAM EUCLID
        PRINT *, 'Find the Greatest Common Divisior (GCD),'
        PRINT *, 'of two numbers A and B using the,'
        PRINT *, 'Euclid Algorithm.'
    1   PRINT *, 'A?'
        READ *, NA
        IF (NA.LE.0) THEN
          PRINT *, 'A must be a positive integer.'
          GOTO 1
        END IF
    2   PRINT *, 'B?'
        READ *, NB
        IF (NB.LE.0) THEN
          PRINT *, 'B must be a positive integer.'
          GOTO 2
        END IF
        PRINT *, 'The GCD of', NA, ' and', NB, ' is', NGCD(NA, NB), '.'
        STOP
      END

      FUNCTION NGCD(NA, NB)
        IA = NA
        IB = NB
    3   IF (IB.NE.0) THEN
          ITEMP = IA
          IA = IB
          IB = MOD(ITEMP, IB)
          GOTO 3
        END IF
        NGCD = IA
        RETURN
      END
